tools/MapTileAggregator/__main__.py

```diff
@@ -0,0 +1,149 @@
+    ("/anticorner/contrasted", 8): frozenset({top_left, bottom_left, bottom_right}),
+    ("/half", 1): frozenset({top_left, bottom_left}),
+    ("/half", 2): frozenset({top_right, bottom_right}),
+    ("/half", 4): frozenset({top_left, top_right}),
+    ("/half", 8): frozenset({bottom_left, bottom_right}),
+    ("/half/contrasted", 1): frozenset({top_left, bottom_left}),
+    ("/half/contrasted", 2): frozenset({top_right, bottom_right}),
+    ("/half/contrasted", 4): frozenset({top_left, top_right}),
+    ("/half/contrasted", 8): frozenset({bottom_left, bottom_right}),
+    ("/opposingcorners", 1): frozenset({bottom_left, top_right}),
+    ("/opposingcorners", 2): frozenset({top_left, bottom_right}),
+    ("", 1): frozenset({top_left}),
+    ("", 2): frozenset({bottom_right}),
+    ("", 4): frozenset({top_right}),
+    ("", 8): frozenset({bottom_left}),
+}
+
+path_dir_sets_reverse = {}
+for k, v in path_dir_sets.items():
+    path_dir_sets_reverse[v] = path_dir_sets_reverse.get(v, []) + [k]
+
+# We don't want to turn "contrasted" into fulltile decals
+compatibility_sets = [
+    {"", "/half/contrasted", "/opposingcorners", "/fourcorners", "/anticorner/contrasted"},
+    {"/half", "/anticorner", "/full"}
+]
+
+compiled_regex = re.compile(r"\/obj\/effect\/turf_decal\/tile\/?([A-Za-z_]+)?([A-Za-z_\/]+)?(\{[\s\S]*(dir = (\d+))[\s\S]*\})?")
+
+def update_map(map_path):
+    the_map = dmm.DMM.from_file(map_path)
+    checked = 0
+    for z, y, x in the_map.coords_zyx:
+        coord = x, y, z
+        tile, turfs, areas = dmm.split_atom_groups(the_map.get_tile(coord))
+        turf_decals = list(filter(lambda atom: atom.startswith('/obj/effect/turf_decal/tile'), tile))
+        if not turf_decals or len(turf_decals) <= 0:
+            continue
+        checked += 1
+        typed = dict()
+        for decal in turf_decals:
+            matched = re.search(compiled_regex, decal)
+
+            if matched == None:
+                print("Warning, what the fuck did we just parse? {}".format(decal))
+
+            color = matched.group(1)
+            if color == None:
+                # Commented out the warning because there's no reason for this to occur. If it exists, it should be addressed case-by-case outside of the scope of this script (with an UpdatePaths)
+                # This is because null-colored turf decals legitimately have no color and don't modify appearances, bit silly. If a null-colored turf decal means something somewhere else, add that color here.
+                # color = "white"
+                # print("Warning, tile with no color (this is bad): {}".format(decal))
+                continue
+
+            last = matched.group(2)
+            if last == None:
+                last = ""
+
+            dir = matched.group(5)
+            if dir == None:
+                dir = 2
+            else:
+                dir = int(dir)
+
+            dirs = None
+            if last == "":
+                dirs = frozenset({ dir })
```

Comment:
You removed the else case here, this is important for parsing out existing stuff that uses non-corners but is still not fully aggregated. Just remove `| tile_side_map`

```
           if last == "":
                dirs = frozenset({ dir })
            else:
                # add in tile_side_map so we can parse it out, but don't use the reverse later
                dirs = frozenset((path_dir_sets | tile_side_map)[(last, dir)])
```

This is also why you had a ton of false positives non the next check, you shouldn't have *any*.

Reply:
You could also just replace the entire block, since the tile_side_map is no longer needed:
```
dirs = frozenset((path_dir_sets)[(last, dir)])
```

No need to check last at all

Reply:
After fixing this you really should rerun the aggregator on a fresh copy of the maps.

Reply:
ah alright thanks for letting me know